lib/iris/common/metadata.py

```diff
@@ -89,25 +98,428 @@ class BaseMetadata(metaclass=_NamedTupleMeta):
 
     __slots__ = ()
 
-    @classmethod
-    def token(cls, name):
+    @lenient_service
+    def __eq__(self, other):
         """
-        Determine whether the provided name is a valid NetCDF name and thus
-        safe to represent a single parsable token.
+        Determine whether the associated metadata members are equivalent.
 
         Args:
 
-        * name:
-            The string name to verify
+        * other (metadata):
+            A metadata instance of the same type.
 
         Returns:
-            The provided name if valid, otherwise None.
+            Boolean.
 
         """
-        if name is not None:
-            result = _TOKEN_PARSE.match(name)
-            name = result if result is None else name
-        return name
+        result = NotImplemented
+        if hasattr(other, "__class__") and other.__class__ is self.__class__:
+            if LENIENT(self.__eq__) or LENIENT(self.equal):
+                # Perform "lenient" equality.
+                logger.debug(
+                    "lenient", extra=dict(cls=self.__class__.__name__)
+                )
+                result = self._compare_lenient(other)
+            else:
+                # Perform "strict" equality.
+                logger.debug("strict", extra=dict(cls=self.__class__.__name__))
+                result = super().__eq__(other)
+
+        return result
+
+    def __lt__(self, other):
+        #
+        # Support Python2 behaviour for a "<" operation involving a
```

Comment:
I believe that the behaviour of "<" changed from Py2 to Py3 and we want to keep the Py2 type behaviour so we implement it explicitly.

Reply:
It is aimed at Python3 only. 

This `__lt__` ensures that the sorting _behaviour_ is consistent across Python versions. I guess we can drop it at some point, but I didn't want to do that right now... simply because there is just sooooo much changed involved in this sequence of PRs - thought that we might save this for another day.

Reply:
I remember looking at this when I was doing work on merge/concat and coming to the conclusion that the Py2 behaviour is actually preferable in this case. I believe it had to do with the default Py3 behaviour throwing errors when None values were present. With that said, there may be a case for rewriting the comment so that it no longer refers to Py2 since it is now less relevant that this behaviour is the same Py2 behaviour.